zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -645,17 +647,29 @@ private void setParagraphMagic(Paragraph p, int index) {
    * Run all paragraphs sequentially. Only used for CronJob
    */
   public synchronized void runAll() {
-    runAll(null, true);
+    String cronExecutingUser = (String) getConfig().get("cronExecutingUser");
+    String cronExecutingRoles = (String) getConfig().get("cronExecutingRoles");
+    if (null == cronExecutingUser) {
+      cronExecutingUser = "anonymous";
+    }
+    AuthenticationInfo authenticationInfo = new AuthenticationInfo();
```

Comment:
Parsing roles here with split(...) does not work anymore, when having more than one role.
There were recent changes to parsing the roles, see:
https://github.com/apache/zeppelin/commit/ad77265c69bf2ea5fe82fa8c15c7a3d3ea2178e3#diff-4a78303414018c7d08a044687ab6afd4R57

This code instead worked locally. AuthenticationInfo constructor will properly parse the roles:
```
    ...
    AuthenticationInfo authenticationInfo = new AuthenticationInfo(
        cronExecutingUser, 
        StringUtils.isEmpty(cronExecutingRoles) ? null : cronExecutingRoles,
        null);
    runAll(authenticationInfo, true);
```

Reply:
Thanks, fixed